tests/Routing/RoutingUrlGeneratorTest.php

```diff
@@ -128,7 +128,7 @@ public function testControllerRoutesWithADefaultNamespace()
         $route = new Illuminate\Routing\Route(['GET'], 'foo/bar', ['controller' => 'namespace\foo@bar']);
         $routes->add($route);
 
-        $route = new Illuminate\Routing\Route(['GET'], 'something/else', ['controller' => 'something\foo@bar']);
+        $route = new Illuminate\Routing\Route(['GET'], 'something/else', ['controller' => '\something\foo@bar']);
```

Comment:
> Can you define "App\Test\Foo@bar" as a route in laravel, and have it working correctly?

How did you do it in order to make it not working, there many packages in Laravel 5+ that uses routes (obviously outside of the root namespace) and working fine.

```php
<?php namespace App\Test;

use Illuminate\Routing\Controller;

class Foo extends Controller
{
    public function bar()
    {
        return action('\App\Test\Foo@bar');
    }
}
```

```php
<?php

namespace App\Providers;

use Illuminate\Routing\Router;
use Illuminate\Foundation\Support\Providers\RouteServiceProvider as ServiceProvider;

class RouteServiceProvider extends ServiceProvider
{
    /**
     * This namespace is applied to the controller routes in your routes file.
     *
     * In addition, it is set as the URL generator's root namespace.
     *
     * @var string
     */
    protected $namespace = 'App\Http\Controllers';

    /**
     * Define your route model bindings, pattern filters, etc.
     *
     * @param  \Illuminate\Routing\Router  $router
     * @return void
     */
    public function boot(Router $router)
    {
        //

        parent::boot($router);
    }

    /**
     * Define the routes for the application.
     *
     * @param  \Illuminate\Routing\Router  $router
     * @return void
     */
    public function map(Router $router)
    {
        $router->group(['namespace' => $this->namespace], function ($router) {
            require app_path('Http/routes.php');
        });

        $router->get('foo', 'App\Test\Foo@bar');
    }
}
```

![screen shot 2015-06-11 at 9 09 19 pm](https://cloud.githubusercontent.com/assets/172966/8107739/2c5ad998-107e-11e5-9c2c-eb393f0e7bb2.png)